browser-extension/src/addVideoStatistics.js

```diff
@@ -1,103 +1,120 @@
 var browser = browser || chrome;
 
-document.addEventListener('yt-navigate-finish', process);
+/**
+ * Youtube doesnt completely load a video page, so content script doesn't
+ * launch correctly without these events.
+ *
+ * This part is called on connection for the first time on youtube.com/*
+ */
+document.addEventListener('yt-navigate-finish', addVideoStatistics);
+
+if (document.body) {
+  addVideoStatistics();
+} else {
+  document.addEventListener('DOMContentLoaded', addVideoStatistics);
+}
 
-if (document.body) process();
-else document.addEventListener('DOMContentLoaded', process);
+/*
+ * Create the statistics container with the tournesol score, comparisons and contributors.
+ */
+function addVideoStatistics() {
+  const videoId = new URL(location.href).searchParams.get('v');
+  if (!location.pathname.startsWith('/watch') || !videoId) return;
 
-/* ********************************************************************* */
+  let videoStatsResponse = null;
+  browser.runtime.sendMessage(
+    {
+      message: 'getVideoStatistics',
+      video_id: videoId,
+    },
+    function (resp) {
+      videoStatsResponse = resp;
+    }
+  );
 
-function process() {
-  // Get video id via URL
-  var videoId = new URL(location.href).searchParams.get('v');
+  const timer = window.setInterval(createStatisticsIsReady, 300);
 
-  // Only enable on youtube.com/watch?v=* pages
-  if (!location.pathname.startsWith('/watch') || !videoId) return;
+  /**
+   * Create the statistics container.
+   */
+  function createStatisticsIsReady() {
+    const actionsRow = document.getElementById(TS_ACTIONS_ROW_ID);
+    if (!actionsRow) return;
 
-  // Timer will run until needed elements are generated
-  var timer = window.setInterval(createButtonIsReady, 300);
-
-  function createButtonIsReady() {
-    /*
-     ** Wait for needed elements to be generated
-     ** It seems those elements are generated via javascript, so run-at document_idle in manifest is not enough to prevent errors
-     **
-     ** Some ids on video pages are duplicated, so I take the first non-duplicated id and search in its childs the correct div to add the button
-     ** Note: using .children[index] when child has no id
-     */
-    if (
-      !document.getElementById('menu-container') ||
-      !document.getElementById('menu-container').children.item('menu') ||
-      !document.getElementById('menu-container').children.item('menu')
-        .children[0] ||
-      !document.getElementById('menu-container').children['menu'].children[0]
-        .children['top-level-buttons-computed']
-    )
-      return;
-
-    // If the button already exists, don't create a new one
-    if (document.getElementById('tournesol-statistics')) {
+    if (document.getElementById('tournesol-statistics-info')) {
       window.clearInterval(timer);
       return;
     }
 
     window.clearInterval(timer);
+    
+    const prev = document.getElementById('tournesol-statistics-info');
+    if (prev) prev.remove();
```

Comment:
There is still an issue here, as the timer could be cleared before the response from the server is received. Causing the "videoNotRatedMessage" to be displayed wrongly. Also, removing `prev` looks unnecessary, as the timer is supposed to be cleared already in this condition.

So I suppose we can keep something like:

```suggestion
    if (videoStatsResponse == null) {
      return;
    }

    window.clearInterval(timer);

    if (document.getElementById('tournesol-statistics-info')) {
      return;
    }
```